packages/tasit-act/src/tasitAct.test.js

```diff
@@ -0,0 +1,83 @@
+import Contract from "./tasitAct";
+import { expect, assert } from "chai";
+// Note: This file is originally genarated by `tasit-contracts` and was pasted here manually
+// See https://github.com/tasitlabs/TasitSDK/issues/45
+import { abi as contractABI } from "./helpers/SimpleStorage.json";
+// Note: Under the current `tasit-contracts` setup SimpleStorage aways will deployed with this address
+// See https://github.com/tasitlabs/TasitSDK/pull/59#discussion_r242258739
+const contractAddress = "0x6C4A015797DDDd87866451914eCe1e8b19261931";
+
+describe.skip("Contract", function() {
+  let simpleStorage;
+
+  beforeEach("should connect to and existing contract", async () => {
+    // Connection with blockchain
+    // Ethers.js allows sets a "connector" (provider or already connected wallet) on constructor of Contract
+    // or later with contract.connect() method.
+    // How connection should be made in here?
+    // - Make sense always uses `tasit-account` to create connected account instead of interact with provider object?
```

Comment:
Hmm, I think the mobile apps built using this SDK will want to be network-data-conscious, so I don't think we should have `tasit-account `create the connection

Reply:
A "signer" will be necessary to send write transactions to contracts, how it'll work without an account/wallet?

Reply:
Ok, got it.

We have three entities here: account(signer), contract and provider.
- Is possible to use `contract` alone, but it is useful only to interact offline with ABI.
- A `contract` only with a `provide` allows read-only interactions with blockchain.
- A `contract` only with an `account` (without `provider`) is usually used for security reasons (sign tx offline and broadcast later). The same for an `account` without a provider.
- A `contract with a `provider` and an `account` allows read and write operations with contracts.

Using `tasit` Decentraland demo as case, `account` can be created later on Ethereum Onboarding screens. But I think that a `provider` should be used since the beginning (to list lands and so on).
So maybe `provider` (without `account`) could be encapsulated by `TasitAct.Contract` and later, when `account` will be created, it could be passed to `contract` to sign transactions.

Ethers.js `provider` is connected to mainnet by default: 

> For most situations, it is recommended that you use a default provider, which will connect to both Etherscan and INFURA simultaneously:

> There are several methods to connect to the Ethereum network provided. If you are not running your own local Ethereum node, it is recommended that you use the getDefaultProvider() method.

I've figured out how it works without API_TOKEN_ID yet.

Is it make sense?

Reply:
(A) That's a perfect description in those four bullet points - thanks!

(B) The way you describe the Decentraland use case is what I'm picturing too. Yeah, so `tasit-act` should be able to get an `account` added to it after it's already been initialized.

(C) The provider is configured to point to mainnet by default? Or it connects (establishes a connection using network data) right away too? If possible we'll want to find a way to configure it to point to mainnet right away through Infura, but not actually send and receive data over the network until we need to. That could end up meaning a PR against ethers.js itself, or maybe we can have it work like that right now - we'll have to look into that.

(D) You mean you haven't figured out how it works without API_TOKEN_ID yet? Okay - I think the end users will have an API_TOKEN_ID, but `tasit-act` will handle passing that through to ethers.js for them.

Reply:
(A) Thanks!
(B) :+1: 
(C) The ethers.js default provider uses infura+etherscan to interact w/ ethereum's mainnet by default. It doesn't connect to a node directly. 
(D) Yes, ethers default provider uses infura without requiring an API_TOKEN_ID. Maybe this kind of public use has a connections rate limit (But I'm not sure).

Reply:
(C) Thanks. Yep, I know it uses those as the full node(s) it points to. This doesn't quite answer my question though. I'm wondering whether data is exchanged over the network right away upon configuring the provider.

Reply:
(D) Oh interesting. Yeah, I bet it does (or will) have a rate limit. Regardless of what other full node someone might opt to use down the road (like if Tasit spun one up for our own use, for instance), it seems reasonable to expect that it will have an API key.

Reply:
(C) Oh, got it now. I don't think so, I think the network will be used when some function is called. I've noted it here to check that.

Reply:
Okay perfect - thanks!

Reply:
I've used some tools (strace, tcpdump) to verify that deeply and using the default provider (infura/etherscan). ethers.js doesn't make any network connection on provider instantiation.